src/hooks/sounds.tsx

```diff
@@ -35,33 +35,40 @@ export interface ExposedDataWithPlay extends ExposedData {
 
 export interface CustomSoundsState {
   soundConfigs: SoundConfigMap;
+  isLoaded: boolean;
 }
 
 export const initialValue: CustomSoundsState = {
   soundConfigs: {},
+  isLoaded: false,
 };
 
 export const CustomSoundsContext = createContext<CustomSoundsState>(
   initialValue
 );
 
 export const CustomSoundsProvider: React.FC = ({ children }) => {
+  const [isLoaded, setIsLoaded] = useState<boolean>(initialValue.isLoaded);
   const [soundConfigs, setSoundConfigs] = useState<SoundConfigMap>(
     initialValue.soundConfigs
   );
 
   // Fetch the sound configs data on first load
   useEffect(() => {
-    fetchSoundConfigs().then((soundConfigs) => {
-      setSoundConfigs(soundConfigs);
-    });
+    fetchSoundConfigs()
+      .then((soundConfigs) => {
+        setSoundConfigs(soundConfigs);
+        setIsLoaded(true);
+      })
+      .catch(() => setIsLoaded(true));
   }, []);
 
   const providerData = useMemo(
     () => ({
       soundConfigs,
+      isLoaded,
```

Comment:
It'd probably be useful to use this new `isLoaded` bool in the `useCustomSound` hook below.

I haven't deeply looked through it/where it would be most useful.. but one place that immediately came to mind was on line 153 within the `useEffect` that has the check `if (hasSoundRef && !hasSoundConfig) {`. We probably don't need to notify Bugsnag here if we haven't even loaded the data yet.

There are almost certainly a few other cases where it would make sense to use within this hook.

Reply:
got it